src/Record.php

```diff
@@ -1437,7 +1437,7 @@ private function buildTables()
                 $sql .= $this->escapeField($this->tables[$i]);
             }
         }
-        $sql .= strlen($this->join_sql) > 0
+        $sql .= strlen($this->join_sql ?? '') > 0
```

Comment:
I don't think it's necessary to make this backward incompatible.  We could change this to
```
$sql .= (isset($this->join_sql) && (strlen($this->join_sql) > 0))
```